### ISSUE TITLE
https://yt-dl.org/ broken by a German court

### ISSUE THREAD

**Issue by DavidGriffith** [ytdl-org/youtube-dl#32495]
## Checklist

- [x] I'm reporting a broken site support
- [x] I've verified that I'm running youtube-dl version **2021.12.17**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar issues including closed ones

## Verbose log
Logs not relevant

## Description

When attempting to download `youtube-dl` as described in `README.md` or `README.txt`, you get a "403 forbidden" error.  Visiting that website gives this message:

----

# Access denied

Due to a [ruling](https://openjur.de/u/2466945.html) of the Hamburg Regional Court, access to this website is blocked.

----
# Zugriff gesperrt

Aufgrund eines [Urteils](https://openjur.de/u/2466945.html) des Landgerichts Hamburg ist der Zugriff auf diese Website gesperrt

**Comment by dirkf** [comment]
This is true though not what "broken site support" was intended to mean.

There's no point downloading that version anyway: the court is actually doing you a favour. When a new release is made the DNS will have been reset to avoid this block.

Meanwhile, refer to #30839 for the nightly build that fixes many known issues.

**Comment by Robin-Velas** [comment]
is there any workaround to download the youtube-dl.exe file?

**Comment by dirkf** [comment]
You don't want to do that. See above.

**Comment by thesouthafrican** [comment]
> is there any workaround to download the youtube-dl.exe file?

I'm by no means advanced, I ended up finding youtube-dl in 2020 and downloading & figuring it out in 2020 out of desperation to download a dropbox video. Also got affected by the "unable to extract uploader ID" error. 
Just figured out how to download the "nightly release" and it's working just fine :) 

Use the [#30839](https://github.com/ytdl-org/youtube-dl/issues/30839) link, download from there, remove the old .exe file and remove its PATH address. 
Then add the new .exe file address to PATH. Close everything (including cmd prompt) and you should be golden :)

**Comment by Stargateur** [comment]
```zsh
git clone https://github.com/ytdl-org/youtube-dl.git
pip install youtube-dl
```

**Comment by Keisial** [comment]
I'm sorry, as you may not like to hear this, but if "There's no point downloading that version", which happens to be the latest release, that's a strong indicator that there should have been a new release since. It's up to you as maintainers, though. While the list of known issues now fixed seem worthy, there may be good reasons not to have cut a release yet.

Anyway, I didn't want my main point to be whether there should be a release or not. I wanted to ask: _How are we expected to know the latest version?_
I used to have:
```
  version="$(wget -q -O - https://yt-dl.org/ | sed -n '/Latest/{ s!.*(v\(....\...\...\)).*!\1!p }')"
```

(which I admit perhaps was not the proper way before, either), but now that obviously fails with a 403 error.

Thank you very much for this software. It's dismaying that it is now having this kind of issues again.

**Comment by fmonts** [comment]
> I'm sorry, as you may not like to hear this, but if "There's no point downloading that version", which happens to be the latest release, that's a strong indicator that there should have been a new release since. It's up to you as maintainers, though. While the list of known issues now fixed seem worthy, there may be good reasons not to have cut a release yet.
> 
> Anyway, I didn't want my main point to be whether there should be a release or not. I wanted to ask: _How are we expected to know the latest version?_ I used to have:
> 
> ```
>   version="$(wget -q -O - https://yt-dl.org/ | sed -n '/Latest/{ s!.*(v\(....\...\...\)).*!\1!p }')"
> ```
> 
> (which I admit perhaps was not the proper way before, either), but now that obviously fails with a 403 error.
> 
> Thank you very much for this software. It's dismaying that it is now having this kind of issues again.

If you really want that version you find the binaries on the github releases page: https://github.com/ytdl-org/youtube-dl/releases/tag/2021.12.17

**Comment by dirkf** [comment]
Even if a version was released with all site extractors passing their tests, the turnover of site issues would invalidate the release after a few days. For now, the nightly release is the best bet. Should some site work with 2021.12.07 and not with the latest nightly build, please open a new issue.

A lot of data about available versions is returned as JSON from https://api.github.com/repos/ytdl-org/youtube-dl/releases/latest. For convenience, the nightly builds are released in a different repo and the internal update check uses the corresponding JSON URL (`s/youtube-dl/ytdl-nightly/`). These version data sets are generated by GH and are disjoint.

**Comment by rcrodrigues** [comment]
For anyone looking for a quick solution:

```
sudo curl -L https://github.com/ytdl-org/youtube-dl/releases/download/2021.12.17/youtube-dl -o /usr/local/bin/youtube-dl
```

**Comment by karstengit** [comment]
The last version can be build with

```
git clone https://github.com/ytdl-org/youtube-dl.git youtube-dl
cd youtube-dl
make youtube-dl
```

But this does not solve the problem that a player must be downloaded from https://yt-dl.org/

Example:
```
youtube-dl -x --audio-format best https://www.youtube.com/watch?v=dCHrO83XEho
[youtube] dCHrO83XEho: Downloading webpage
[youtube] dCHrO83XEho: Downloading API JSON
WARNING: unable to extract player URL; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
```

Is it possible to integrate the needed player here at github?
So that the player is integrated in the youtube-dl ?

**Comment by dirkf** [comment]
https://github.com/ytdl-org/youtube-dl/issues/32495#issuecomment-1666729299

**Comment by karstengit** [comment]
> the court is actually doing you a favour.

Censoring as favour is a nice idea. ;-)

A traceroute shows
```
 5  ipv4.de-cix.fra.de.as25560.rh-tec.de (80.81.193.106)  19.162 ms  18.317 ms  19.692 ms
 6  * * *
 7  62.216.183.46 (62.216.183.46)  14.936 ms  13.716 ms  14.057 ms
 8  aries.uberspace.de (95.143.172.170)  14.053 ms !X  13.979 ms !X  14.604 ms !X
```

The IP is blocked by this company:
**rh-tec Business GmbH**
(DCH) Data Center/Web Hosting/Transit
https://www.ip2location.com/demo/95.143.172.170

Professional censoring in germany!
https://www.ip2location.com/as25560

**Comment by neszt** [comment]
Why is there no new release? Why is the last one in 2021? If it cannot be solved so that the last version can be downloaded, then why don't you write that "Youtube-dl cannot be downloaded"? Why is there a link out there that points to an "Access Denied" page?

**Comment by dirkf** [comment]
https://github.com/ytdl-org/youtube-dl/issues/32495#issuecomment-1666729299 still applies.

**Comment by PSLLSP** [comment]
I am in Central Europe (not in Germany) and the site is blocked in our country:
![yt-dl-blocked](https://github.com/ytdl-org/youtube-dl/assets/6446102/65d03f5b-3ac5-4a2a-9eb3-08a2d67523ba)

This is the first time I see this kind of censorship...

Description on github has instructions to use that site, but it doesn't work (because the site is blocked)

```
user@ubuntu:~/tmp$ curl -L https://yt-dl.org/downloads/latest/youtube-dl -o youtube-dl
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   384  100   384    0     0   4042      0 --:--:-- --:--:-- --:--:--  4042

user@ubuntu:~/tmp$ ll youtube-dl; file youtube-dl 
-rw-rw-r-- 1 user user 384 Jan  9 20:15 youtube-dl
youtube-dl: HTML document, ASCII text
```

**Comment by dirkf** [comment]
Until the domain points to a valid domain, a version of the README that doesn't mention yt-dl.org is [provided in the PR above](https://github.com/ytdl-org/youtube-dl/blob/9324d03ef9cfd8e26874cead3a2c1b5c78cdb42c/README.md).

**Comment by rugk** [comment]
BTW it properly responds with:
> 451 Unavailable For Legal Reasons

I like that! :smile:

**Comment by facundoq** [comment]
Dirk, I'm sorry for you for all this german court nonsense. 

But it's confusing as hell having a program known as "youtube-dl" whose readme on the main branch points to a 404 link, *but* there's _another_ readme that has the corrected links, *but* you don't want that anyway because it's an old version that won't even work with any youtube video.

 Just point to the latest release in the main readme and be done with it..
